pkg/cloudprovider/aws/instancetype.go

```diff
@@ -92,25 +92,38 @@ func (i *InstanceType) Resources() v1.ResourceList {
 	return i.resources
 }
 
+// Price will get the minimum price of all offerings based on the passed filter
+// so that we can use the cheapest offerings at scheduling time
+func (i *InstanceType) Price(filters ...func(o cloudprovider.Offering) bool) float64 {
```

Comment:
I don't think we need this `Price` func anymore since pricing is on Offerings and that is already exposed via the CloudProvider interface.  Rather than doing this filtering, I think the consolidation logic can do the offering parsing itself outside of the cloud provider. If it's shifted outside of the cloud provider, it will make onboarding new cloud providers easier, shifting complexity to the core.